libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c

```diff
@@ -725,6 +725,8 @@ static OTA_Err_t prvStartHandler( OTA_EventData_t * pxEventData )
     OTA_Err_t xReturn = kOTA_Err_None;
     OTA_EventMsg_t xEventMsg = { 0 };
 
+    /* Start self-test timer, if platform is in self-test. */
+    prvStartSelfTestTimer();
```

Comment:
Can you explain why we moved the timer start?

Reply:
The timer should be started even in case device can not get a job document or OTA job is forced cancelled. This helps to ensure that in all cases timer is started and device can roll back if the self-test is not completed.